### ISSUE TITLE
Bindings on the page won't refresh after the XHR/fetch completes in Firefox

### ISSUE THREAD

**Issue by Freak613** [angular/angular#9385]
**I'm submitting a ...**

```
[x] bug report
[ ] feature request
[ ] support request
```

**Current behavior**
Bindings on the page won't refresh after the XHR/fetch completes.

**Reproduction of the problem**
Call fetch and try to display something when the loading complete. Please, see an example here:
https://github.com/Freak613/quickstart/blob/master/app/app.component.ts
It't just quickstart app with simple fetch loading of single json file.

Browser: Firefox.

**What is the expected behavior?**
The page should display the json: '{data:1}'

**Please tell us about your environment:**
- **Angular version:** 2.0.0-rc.1, 2.0.0-rc.2
- **Browser:** [Firefox 46.0]
- **Language:** [TypeScript] 

We've updated our project to the angular 2.0.0-rc.1.
What I've found out by trial and error:

In Firefox:
1. NgZone.onStable and NgZone.onUnstable do not occur after the XHR/fetch completes.
2. We've managed to work around this issue by excluding core-js/es6/promise. So, I suggest that this polyfill breaks change detection in Firefox.
3. If we call changeDetector.detectChanges() then Firefox updates view with recieved data.

**Comment by gdi2290** [comment]
this is a problem with either the promise polyfills in core-js or a conflict between that and firefox.
you need to import everything but the promise polyfill
https://github.com/AngularClass/angular2-webpack-starter/blob/master/src/polyfills.browser.ts#L6...L26
related https://github.com/AngularClass/angular2-webpack-starter/issues/709

**Comment by Freak613** [comment]
I think this is not a good fix of this issue. Probably it may indicate of some library compability problems. I've unsuccessfully tried to debug it by myself but I can't find what is the root of the issue: angular, core-js, zone-js or firefox and what should we do to fix it. That's why I didn't create issue in core-js project - I'm not sure is it their problem. core-js is quite popular library and it's surprising that anybody till now hasn't detect issues with core-js Promise.

**Comment by gdi2290** [comment]
this could also be a zone.js conflict with either firefox or core-js promise
cc @mhevery

**Comment by vicb** [comment]
@Freak613 could you please provide a minimal repro in a plunker (try to reproduce with only zone + polyfill and add angularonly if strictly required). Thanks for your help.

**Comment by Freak613** [comment]
I can't reproduce this problem with only zone or NgZone. It can be reproduced only with angular:
http://plnkr.co/edit/LUMTAlPgqlbS1kXHSkQ6?p=preview

**Comment by mhevery** [comment]
The issue is with `core-js` and FireFox. Specifically:

1. `core-js` tries to polyfill `Promise`. The check it uses is [here](https://github.com/zloirock/core-js/blob/master/modules/es6.promise.js#L25). Somehow this check fails on FireFox, and FireFox gets a Promise polyfill even thought FireFox has native `Promise`
2. `zone.js` loads and creates its own Promise polyfill regardless if there is native one. Zone also monkey patches `Zone.prototype.then` with Zone magic. (But it is `core-js` `Promise` rather then native `Promise`.)
3. Angular application runs and invokes `fetch()` which always returns native `Promise`. Invoking `then` then is not intercepted by the Zone (because Zone monkey patched the `core-js` `Promise`)

Solution: Fix `core-js` to correctly detect FireFox `Promise`

Possible Workarounds: 
 
1. Load `Zone` before `core-js`
2. Load a polyfill which only has ES6 collections rather then the whole world.

**Comment by gdi2290** [comment]
cc @zloirock

**Comment by zloirock** [comment]
Sorry, I still hadn't time to completely explore the problem.

@mhevery in `core-js` FF `Promise` correctly  detected as not supported because FF does not support unhandled rejection tracking, so by your exploration the problem in incorrect  `zone.js ` behavior.

**Comment by rhmoller** [comment]
I am experiencing the same problem with RC6

I use these polyfills

```
require("core-js/es6");
require("core-js/es7/reflect");
require("zone.js/dist/zone");
import "whatwg-fetch";
````

If I load zone before core-js, as suggested by @mhevery it also breaks in Chrome and still breaks in Firefox

**Comment by DzmitryShylovich** [comment]
is this still an issue with the latest angular/zones/core-js?

**Comment by Freak613** [comment]
I can confirm that with updating zone.js from 0.6.12 to 0.7.2 issue has been fixed in Firefox@50, even with angular@2.0.0-rc.3
http://plnkr.co/edit/lo56jzIvPJNsDnbl3KDt?p=preview

Also, works ok in Chrome@55 and Safari@10(with fetch polyfill)

Hands up for one more mysteriously disappeared bug!

**Comment by Freak613** [comment]
Issue has been fixed in zone.js@0.6.19, but it's not clear how.
